packages/terra-date-picker/src/react-datepicker/day.jsx

```diff
@@ -214,12 +214,23 @@ class Day extends React.Component {
       this.props.month !== getMonth(this.props.day)
   }
 
+  isBorderApplied = () => {
+    if(this.isSameDay(this.props.preSelection) && !isDayDisabled(this.props.selected, this.props) 
+    && (document.activeElement.hasAttribute('data-terra-open-calendar-button') 
```

Comment:
If there are multiple date-pickers on the page, is there any risk that this will apply the border to both?